### PR TITLE
Fix for LightUp and TreeTent Puzzle Creators

### LINKED ISSUE #372: [ENHANCEMENT] Add Support for LightUp in Puzzle Creator
### Describe the current behavior of what you're trying to improve. If your enhancement request related to a problem, please also describe the problem.

Currently, we only have puzzle creation support for Nurikabe and TreeTent

### Describe the improvement you'd like

We would like to introduce puzzle creation support for LightUp. Since this has already been done twice, there should be a fair amount of reference code to assist.

The non-placeable elements should be:
- Black square
- Black square with number
- Yellow square (lit space)
- Gray square (unlit space)

The placeable elements should be:
- Lightbulb
- Dot (asserts no lightbulb can be in space)

Note, the dots should be able to overlap yellow spaces

### Describe alternatives you've considered

_No response_

### Additional Context

See the Nurikabe editor and TreeTent editor for examples

### PR DESCRIPTION
**Pull request by rspacerr**
## Description

This should fix both the LightUp and TreeTent puzzle creators, with functionality to place objects in the creator, save the puzzle and load it in the puzzle editor. 

Closes #372 and #503

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Enhancement (improvement to an already existing feature)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

I opened LightUp and TreeTent independently in the puzzle creator, placed objects, and saved the file, which loaded in the puzzle editor. Fair warning, I was unable to test in the new gradle version since for some reason my build errors out which I am trying to fix. Warning 2, I found that closing the file using ctrl-q from the puzzle editor and loading into the puzzle creator causes a bug, which I think is a native error since this happens for Nurikabe too. I can make an issue after I fix my gradle.

## Checklist:

- [X] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (if applicable)
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by Corppet — CHANGES_REQUESTED** [review]
Please look at the files you are changing in this pull request. There are a lot of redundant changes that need to be addressed before I can approve/review anything here.

**Review by Corppet — CHANGES_REQUESTED** [review]
Changes are looking a lot better than before, though your changes are causing some tests in `TreeTent` to fail. Look in the `checkstyle` check for more details.

**Comment by charlestian23** [comment]
I seem to have fixed those issues Ivan mentioned in a separate pull request (#542). Seems like updating from `dev` didn't bring those changes over?

File: src/main/java/edu/rpi/legup/puzzle/lightup/elements/UnknownTile.java
Comment (charlestian23) [inline src/main/java/edu/rpi/legup/puzzle/lightup/elements/UnknownTile.java:1]: We should not have Nurikabe IDs and images be used for Light Up puzzles. You should make a brand new ID and image for the Light Up unknown tile.

**Comment by Corppet** [comment]
This PR is still a work in progress as not all the checks are successful. If you would like to contribute to this PR you can checkout the branch and try building the project locally to see the errors.

**Comment by jac-oblong** [comment]
PR replaced by #629 and #692

**Pull request closed without merging**